artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -2983,6 +2988,18 @@ else if (consumerRate < threshold) {
                   if (connection != null) {
                      ActiveMQServerLogger.LOGGER.slowConsumerDetected(serverConsumer.getSessionID(), serverConsumer.getID(), getName().toString(), connection.getRemoteAddress(), threshold, consumerRate);
                      if (policy.equals(SlowConsumerPolicy.KILL)) {
+                        if (connection instanceof CoreRemotingConnection) {
```

Comment:
This should be abstract through the ProtocolManagers somehow. You can't do a protocol specific implementation on QueueImpl.